### PR TITLE
Pagination fixes

### PR DESCRIPTION
**Pull request by nateyazdani**
Fix broken `limit` query argument behavior, correct outdated docstring, and load maximum and default rows per page from the configuration dictionary.

### CODE REVIEW

File: statscache/app.py [hunk 1]
```diff
@@ -39,16 +38,22 @@ def paginate(queryset):
...
-    queryset = \
-        queryset.offset((page_number - 1) * page_length).limit(page_length)
+    page_start = (page_number - 1) * page_length
+    page_stop = min(page_length, items_count - page_start)
+    queryset = queryset.slice(page_start, page_stop)
+
+    if page_start > items_count:
+        # In this case, an empty response is safely generated, but it would be
+        # bad practice to respond to invalid requests as if they were correct.
+        flask.abort(400)
```
Comment (ralphbean) [inline statscache/app.py:56]: It's a shame it's reserved for WebDav, but http code 422 would be neat to use here.  _c'est la vie_.

**Comment by ralphbean** [comment]
Looks good to me (this is the first time I've seen that `query.slice(..)` call.  I usually use `query.offset(..)` and `query.limit(..)`, but I quite like the way .slice(..) looks.

**Comment by ralphbean** [comment]
If that wasn't clear -- :+1:

**Comment by nateyazdani** [comment]
Thanks! I stumbled upon the `query.slice(a, b)` method by accident when I was figuring what was happening with `query.limit()`. `query.slice(a, b)` is actually just equivalent to `query.offset(a).limit(b)`, but like you said, it does look pretty fancy ;)

**Pull request merged**